qemu.go

```diff
@@ -286,6 +286,30 @@ func (q *qemu) appendBlockDevice(devices []govmmQemu.Device, drive Drive) []govm
 	return devices
 }
 
+func (q *qemu) appendVhostUserDevice(devices []govmmQemu.Device, vhostUserDevice VhostUserDevice) []govmmQemu.Device {
```

Comment:
Having this function and having different types for each vhost-user type means you have two options. Implement an interface or have multiple functions. Or send in the type as an interface and perform a type switch internally.

Reply:
I'll end up moving to using a VhostUserDevice base type, and switch on the subtype (VhostUserNetDevice, VhostUserSCSIDevice, etc).